### DISCUSSION TITLE
How to open a specific CHM-Page

### DISCUSSION THREAD

**Discussion by Mae72DE** [sumatrapdfreader/sumatrapdf#3572] [Q&A]
What do I have to do to open a specific page of a CHM document?

For example:

`"%ProgramFiles%\SumatraPDF\SumatraPDF.exe" "%ProgramFiles%\7-Zip\7-zip.chm" "its://1/cmdline/syntax.htm"`

**Comment by GitHubRulesOK** [comment]
Hmm chm is an odd one as there are 2 possible states 

so when using inbuilt windows it only accessible by page number which is a different page number to other state 

`SumatraPDF-3.4.6-32 -page 19 7-zip.chm`
![image](https://github.com/sumatrapdfreader/sumatrapdf/assets/12754913/075026f2-afe1-4a07-ac45-4d19a7fb4a53)

However in Fixed UI 
```
ChmUI [
	UseFixedPageUI = true
]
```
you can use  `-page 27` ! OR
`SumatraPDF-3.4.6-32 -named-dest "Command Line Syntax" 7-zip.chm` (or simply `-named-dest Syntax`)
![image](https://github.com/sumatrapdfreader/sumatrapdf/assets/12754913/a8bd8de8-a8ce-4bb4-a486-b75e4a304a69)

With 3.4 and pre-release there are other options but BEWARE search is more likely to cause a crash with Windows HH   
so do NOT use `UseFixedPageUI = false` and -search 
`SumatraPDF-prerel -search "Syntax" 7-zip.chm`  would in this case work for fixed UI mode
![image](https://github.com/sumatrapdfreader/sumatrapdf/assets/12754913/43907560-25ff-4f90-849e-7d44f0753c04)

**Reply by lesensei** [reply]
Unfortunately, that doesn't cover all use cases.
I am currently trying to display the help from KeePass in sumatrapdf. Opening the help file itself is no trouble, and navigating it works as expected.
However, if I want KeePass to open contextual help pages (for example, the help for the "edit entry" dialog), the only thing I can use in the command-line is the path of the specific html page inside the chm file: `v2/entry.html`. I can't even use the term "entry" with the `-named-dest` option since the localized web page doesn't use it.
Is there any other way I can proceed ?

**Reply by kjk** [reply]
You can open an issue with a feature request, provide a test file and explain how you would like this to work.

I can extend `-named-dest` logic for .chm files but I need the above info.

**Reply by lesensei** [reply]
Here it is: #3904 

Thank you for your time !
